src/shaders/symbol_icon.vertex.glsl

```diff
@@ -56,7 +60,11 @@ void main() {
     float anchorZ = a_z_tile_anchor.x;
     vec2 tileAnchor = a_z_tile_anchor.yz;
     vec3 h = elevationVector(tileAnchor) * elevation(tileAnchor);
-    vec4 projectedPoint = u_matrix * vec4(vec3(a_pos, anchorZ) + h, 1);
+    vec3 world_pos = mix_globe_mercator(
```

Comment:
Just a tiny improvement idea: since the `mix_globe_mercator` function is called two times in these shaders, we could spare most of the operations if we only do the `mercator_tile` calculation inside that function, and we do the interpolation here. (Because if I see only the 3rd parameter is different between the calls.)

Reply:
Yes that's a good point. There should be ways to rearrange that to reduce the number of redundant operations. I think there's also a remainder `pow` instruction that only depends on a uniform, which we could move to the CPU. Looking into that.

Reply:
Updated in https://github.com/mapbox/mapbox-gl-js/commit/55418d5389299c2a8f8bee77eab05d9e8d4fe05a and backported to the dev branch.